dsgrid/units/energy.py

```diff
@@ -100,23 +107,157 @@ def to_mbtu(unit_col, value_col):
         .when(F.col(unit_col) == TWH, (F.col(value_col) / KWH_PER_MBTU / 1_000_000_000))
         .when(F.col(unit_col) == THERM, (F.col(value_col) / THERMS_PER_MBTU))
         .when(F.col(unit_col) == MBTU, F.col(value_col))
+        .when(F.col(unit_col) == "", F.col(value_col))
         .otherwise(None)
     )
 
 
-def convert_units(df, columns, from_records, from_to_records, to_unit_records) -> DataFrame:
+def from_any_to_any(from_unit_col, to_unit_col, value_col):
+    """Convert a column based on from/to columns."""
+    return (
+        F.when(F.col(from_unit_col) == F.col(to_unit_col), F.col(value_col))
+        .when(F.col(from_unit_col) == "", F.col(value_col))
+        # From KWh
+        .when(
+            (F.col(from_unit_col) == KWH) & (F.col(to_unit_col) == MWH), F.col(value_col) / 1_000
```

Comment:
I recommend making more constants to use here to convert between SI prefixes (K, M, G, T) and keep everything in muplication as much as you can (e.g., create THERM_PER_KWH in addition to KWH_PER_THERM and just do that division there), because we'll have to do this many times over when other units are introduced.